File: LeetCode/py/1678.py
#Link: https://leetcode.com/problems/goal-parser-interpretation/

class Solution:
    def interpret(self, command: str) -> str:
        result = ""
        for i in range(len(command)):
            if command[i] == "G":
                result += "G"
            elif command[i] == "(":
                if command[i+1] == ")":
                    result += "o"
                else:
                    result += "al"
        return result
        

File: LeetCode/cpp/1678.cpp
//Link: https://leetcode.com/problems/goal-parser-interpretation/

class Solution {
public:
    string interpret(string command) {
        string result = "";
        for(int i = 0 ; i < command.length() ; i++) {
            if (command[i] == 'G')
                result += 'G';
            else if (command[i] == '(') {
                if(command[i+1] == ')')
                    result += 'o';
                else
                    result += "al";
            }
        }
        return result;
    }
};
